libraries/plugins/peerplays_sidechain/peerplays_sidechain_plugin.cpp

```diff
@@ -73,23 +70,87 @@ void peerplays_sidechain_plugin::plugin_set_program_options(
    cfg.add(cli);
 }
 
-void peerplays_sidechain_plugin::plugin_initialize(const boost::program_options::variables_map& options)
+void peerplays_sidechain_plugin_impl::plugin_initialize(const boost::program_options::variables_map& options)
 {
-   ilog("peerplays sidechain plugin:  plugin_initialize()");
+   config_ready_son = options.count( "son-id" ) && options.count( "peerplays-private-key" );
+   if (config_ready_son) {
+   } else {
+      wlog("Haven't set up SON parameters");
```

Comment:
I think we should exit the node here by throw.

Reply:
Agree...